File: backend/src/main/java/com/project2/domain/rank/enums/RankingPeriod.java
package com.project2.domain.rank.enums;

import java.time.LocalDateTime;

public enum RankingPeriod {
    ONE_MONTH(1),
    THREE_MONTHS(3),
    SIX_MONTHS(6);

    private final int months;

    RankingPeriod(int months) {
        this.months = months;
    }

    public LocalDateTime getStartDate() {
        return LocalDateTime.now().minusMonths(months);
    }
}

File: backend/src/main/kotlin/com/project2/domain/rank/enums/RankingPeriod.kt
package com.project2.domain.rank.enums

import java.time.LocalDateTime

enum class RankingPeriod(private val months: Int) {
    ONE_MONTH(1),
    THREE_MONTHS(3),
    SIX_MONTHS(6);

    fun getStartDate(): LocalDateTime {
        return LocalDateTime.now().minusMonths(months.toLong())
    }
}